### DISCUSSION TITLE
strange keybinding alt+shift+plus doesn't show up in macos app menu

### DISCUSSION THREAD

**Discussion by networkhermit** [ghostty-org/ghostty#4007] [Q&A] [answered]
Under Terminal Inspector, `alt shift =` generates `alt+shift+plus` and `+` generates `shift+plus`. This behavior surprised me. I don't know if this is expected.

---

And, might due to the strange sequence of `alt+shift+plus`, it can't be displayed in the Menu `Split Right`.

```text
keybind = alt+shift+minus=new_split:down
keybind = alt+shift+plus=new_split:right
```

<img width="219" alt="image" src="https://github.com/user-attachments/assets/2aa114a3-7986-4be0-9ddc-60bdfb898b42" />

Although `alt+shift+equal` is an invalid keybinding, it actually showed up in the macos app menu as expected.

**Comment by rhodes-b** [comment]
on linux it also seems that pressing `shift =` also generates `shift+plus` keycode I looked into this a bit and the key is passed in from GTK which ghostty has no control over which makes this seem like something at the OS level that pressing `shift =` generates the `+` key

Looking at the physical key GTK reports it as the equal key (this is separate from the other key GTK sends which would match whatever your layout is defined to send)

At least it is consistent between Linux and Mac

but this still seems odd to me since I cant find any other keys that exhibit this behavior shift is generally a modifier and you have to look up the raw keycode plus the modifiers to typically get the correct keycode but it just seems that the original keycode getting passed in is wrong only for equal

edit: I just tried every key on the keyboard with shift only `equal` seems to exhibit this behavior

**Reply by networkhermit** [reply]
Windows Terminal also uses `alt+shift+plus` sequence. So this might just the way it works. I will update the title to reflect the remaining macos menu bug.

```text
{ "keys": "alt+shift+-", "id": "Terminal.DuplicatePaneDown" },
{ "keys": "alt+shift+plus", "id": "Terminal.DuplicatePaneRight" }
```

https://learn.microsoft.com/en-us/windows/terminal/customize-settings/actions#split-a-pane

**Comment by mendoza** [comment]
I was trying to do something similar, but I can't use my MBP keyboard plus (shift + =)

this is the key bind I was trying to achieve (also based on Windows terminal and some tmux)
```
keybind=ctrl+a>+=new_split:right
```

I found a way to do it by doing
```
keybind=ctrl+a>shift+plus=new_split:right
```

**Reply by qmn1711** [reply]
Thanks for your discovery, @mendoza!

I spent so much time trying to figure out why `keybind = ctrl+b>R=reload_config` wasn't working. After your comment, it should be `keybind = ctrl+b>shift+r=reload_config`

This tricky problem should be mentioned in the Trigger Sequences doc.

**Comment by mitchellh** [comment, marked as answer]
Keybindings have changed significantly since 1.0. Please try this on a new nightly release and open a new issue if this still happens. Thank you.
